pyrealm_build_data/synth_data.py

```diff
@@ -0,0 +1,81 @@
+"""This script uses a parametrized model to compress the input dataset.
+
+It fits a time series model to the input data and stores the model parameters.
+The dataset can then be reconstructed from the model parameters using the `reconstruct`
+function, provided with a custom time index.
+"""
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+VAR_BOUNDS = dict(
+    temp=(-25, 80),
+    patm=(3e4, 11e4),
+    vpd=(0, 1e4),
+    co2=(0, 1e3),
+    fapar=(0, 1),
+    ppfd=(0, 1e4),
+)
+
+
+def make_time_features(t: np.ndarray) -> pd.DataFrame:
+    """Make time features for a given time index."""
+    dt = pd.to_datetime(t).rename("time")
+    df = pd.DataFrame(index=dt).assign(const=1.0)
```

Comment:
Can you explain the features.

Reply:
The model approximates the time series by a function $f(t) = a_0 + a_1 t + \sum_f sin(2\pi f t) + cos(2\pi f t)$, where $f = 730.5, 365.25, 12, ...$. Each feature is just a term in this function. I can add this into a comment.